waterdata/location.py

```diff
@@ -0,0 +1,173 @@
+            longitude (str): decimal longitude
+
+        :param: dict monitoring_location_record: a record for a site from an RDB file; the dictionary must contain the
+        following keys at a minimum: site_no, agency_cd, station_nm, dec_lat_va, dec_long_va
+
+        """
+        for key, value in monitoring_location_record.items():
+            setattr(self, '_{}'.format(key), value)
+        self.location_number = getattr(self, '_site_no')
+        self.agency_code = getattr(self, '_agency_cd')
+        self.location_name = getattr(self, '_station_nm')
+        self.latitude = getattr(self, '_dec_lat_va')
+        self.longitude = getattr(self, '_dec_long_va')
+        self._resp = None
+        self._site_param_info = None
+
+    def _get_location_parameter_info(self):
+        """
+        Perform a web service call get basic information
+        about a site and the types of data available at the
+        site.
+
+        :return: records of they types of data at the site
+        :rtype: list
+
+        """
+        params = {'format': 'rdb',
+                  'sites': self.location_number,
+                  'seriesCatalogOutput': True,
+                  'siteStatus': 'all',
+                  'agencyCd': self.agency_code
+                  }
+        resp = execute_get_request(self.service_endpoint, '/nwis/site/', params)
+        if resp.status_code != 200:
+            return resp, []
+        else:
+            parsed_rdb = parse_rdb(resp.iter_lines(decode_unicode=True))
+        return resp, list(parsed_rdb)
+
+    def get_location_metadata(self, expanded=False):
+        """
+        Get the expanded metadata for a site.
+
+        Note that the 'siteOutput' query parameter
+        cannot be used with the 'seriesCatalogOutput' parameter.
+        So, we have to make another webservice call.
+
+        :return: response object and expanded metadata
+        :rtype: tuple
+
+        """
+        params = {'format': 'rdb',
+                  'site': self.location_number,
+                  'agencyCd': self.agency_code
+                  }
+        if expanded:
+            params['siteOutput'] = 'expanded'
+        resp = execute_get_request(self.service_endpoint, '/nwis/site/', params)
+        if resp.status_code != 200:
+            return resp, {}
+        else:
+            parsed_rdb = parse_rdb(resp.iter_lines(decode_unicode=True))
+            expanded_data = next(parsed_rdb)
+        return resp, expanded_data
+
+    def get_capabilities(self):
+        """
+        Determine the parameters measured at the site.
+
+        :return: USGS parameter codes measured at a site
+        :rtype: set
+
+        """
+        if (self._resp is None or self._site_param_info is None) or self._resp.status_code != 200:
+            self._resp, self._site_param_info = self._get_location_parameter_info()
+        supported_params = set([timeseries['parm_cd'] for timeseries in self._site_param_info])
```

Comment:
You can pass an iterator to set here, and avoid creating an intermediate list: `set(timeseries['parm_cd'] for timeseries in self._site_param_info)`